### PR TITLE
Added SetLogHandler method on client.

### LINKED ISSUE #48: Allow to log raw I/O in Client
Like Server.Debug but for Client.

### PR DESCRIPTION
**Pull request by discobean**
I believe it fixes #48

I just wanted a way to specify more detailed logging.

Use it by 
```
c.SetLogHandler(func(message string) {
  log.Print(message)
})
```

### CODE REVIEW

**Comment by foxcpp** [comment]
For referenced issue I had the idea of exactly replicating `Debug io.Writer` of server code by wrapping net.Conn with io.TeeWriter. Do you think callback is a better approach?

**Comment by discobean** [comment]
Most definitely replicating io.Writer is better, but this was just a quick way to get it over the line for me.

File: go.mod [hunk 1]
```diff
@@ -1,5 +1,8 @@
 module github.com/emersion/go-smtp
 
-require github.com/emersion/go-sasl v0.0.0-20200509203442-7bfe0ed36a21
+require (
+	github.com/emersion/go-sasl v0.0.0-20200509203442-7bfe0ed36a21
+	github.com/sirupsen/logrus v1.6.0 // indirect
```
Comment (emersion) [inline go.mod:5]: We don't need this dep

**Comment by emersion** [comment]
I'd prefer an `io.Writer` as well.

**Pull request closed without merging**